README.md

```diff
@@ -1,126 +1,188 @@
+| 	description       | string  | false      |  specify the description of the variable.|
+| 	type              | string  | false      |  default to `string` if not specified (current supported types are string, multiline, boolean, int, enum, password, storageclass, hostname, pvc, and secret).|
+| 	required          | bool    | false      |  define if the variable is required or not (true \| false)|
+| 	default           | string  | false      |  specify the default value. |
+| 	group             | string  | false      |  group questions by input value. |
+| 	min_length        | int     | false      | min character length.|
+| 	max_length        | int     | false      | max character length.|
+| 	min               | int     | false      |  min integer length. |
+| 	max               | int     | false      |  max integer length. |
+| 	options           | []string | false     |  specify the options when the vriable type is `enum`, for example: options:<br> - "ClusterIP" <br> - "NodePort" <br> - "LoadBalancer"|
+| 	valid_chars       | string   | false     |  regular expression for input chars validation. |
+| 	invalid_chars     | string   | false     |  regular expression for invalid input chars validation.|
+| 	subquestions      | []subquestion | false|  add an array of subquestions.|
+| 	show_if           | string      | false  | show current variable if conditional variable is true, for example `show_if: "serviceType=Nodeport"` |
+| 	show\_subquestion_if |  string  | false     | show subquestions if is true or equal to one of the options. for example `show_subquestion_if: "true"`|
+
+**subquestions**: `subquestions[]` cannot contain `subquestions` or `show_subquestions_if` keys, but all other keys in the above table are supported. 
+
+### 6. Test your changes
+
+#### Generate modified chart
+To test your modified chart you will first need to generate it. The result is a tarball containing your upstream chart 
+with patch changes applied, and overlay files added on top of it. It will be located in the `assets` directory and its
+name will be the same as the upstream chart, but the value of `packageVersion` from `package.yaml` is appended to
+it as following `{CHART_NAME}-{VERSION}{packageVersion}.tgz`. 
+
+This will validate and generate a tarball of your modified chart
+```
+make validate CHART={CHART_NAME}
+```
 
-`make charts`: 
+Alternatively, this will skip validation and just generate the tarball of your modified chart (Validating it before submitting a pull request is recommended)
+```
+make charts CHART={CHART_NAME}
+```
 
-Generate tarball for each charts. Use `CHART=${NAME}` for specific chart.
+#### Test modified chart
 
-`make patch`: 
+An easy way to test your chart is by hosting the `assets` directory (Make sure you generate the tarball first) using 
+`python -m SimpleHTTPServer`, expose it using `ngrok`, and use the `ngrok` URL to add the chart as a custom catalog in Rancher.
```

Comment:
Potentially easier way is to just `helm install assets/x.tgz` with the kubeconfig set to that cluster. Not sure which route we want them taking, I guess they don't get questions.yaml that way.

Reply:
Is important they can test their chart with the questions.yaml and app-readme.md included because that's usually where new partners need help the most.